source/NanostackRfPhySpirit1.cpp

```diff
@@ -611,17 +611,15 @@ static void rf_ack_loop(void) {
 
     do {
         /* Wait for signal */
-        osEvent event = rf_ack_sender->signal_wait(0);
+        uint32_t signals = ThisThread::flags_wait_any(RF_SIG_ACK_NEEDED | RF_SIG_CB_TX_DONE | RF_SIG_CB_RX_RCVD);
 
-        if(event.status != osEventSignal) {
+        if(!(signals & (RF_SIG_ACK_NEEDED | RF_SIG_CB_TX_DONE | RF_SIG_CB_RX_RCVD)) {
```

Comment:
Is this check really necessary considering that the previous line explicitly waits for any of those signals?

Reply:
It should only occur in the event of some sort of OS error - it's the equivalent of the test it's replacing. I decided to leave it for the "no functional change" claim.